--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -73,6 +73,7 @@ if (BUILD_SHARED_LIBS AND BUILD_TESTING)
 endif ()
 option (BUILD_PROGRAMS "Build programs" ON)
 option (BUILD_EXAMPLES "Build examples" ON)
+option (ENABLE_DOCS "Allow install related library documentations" ON)
 option (ENABLE_CPACK "Enable CPack support" ON)
 option (ENABLE_BOW_DOCS "Enable black-on-white html docs" OFF)
 if (MSVC AND (DEFINED ENABLE_STATIC_RUNTIME))
@@ -889,32 +890,36 @@ if (INSTALL_MANPAGES)
 	install (FILES man/sndfile-interleave.1 DESTINATION ${CMAKE_INSTALL_MANDIR}/man1 RENAME sndfile-deinterleave.1)
 endif ()
 
-if (ENABLE_BOW_DOCS)
-	set (HTML_BGCOLOUR "white")
-	set (HTML_FGCOLOUR "black")
-else ()
-	set (HTML_BGCOLOUR "black")
-	set (HTML_FGCOLOUR "white")
+if (ENABLE_DOCS)
+
+	if (ENABLE_BOW_DOCS)
+		set (HTML_BGCOLOUR "white")
+		set (HTML_FGCOLOUR "black")
+	else ()
+		set (HTML_BGCOLOUR "black")
+		set (HTML_FGCOLOUR "white")
+	endif ()
+	set (dist_doc_DATA
+		docs/index.md
+		docs/libsndfile.jpg
+		docs/libsndfile.css
+		docs/print.css
+		docs/api.md
+		docs/command.md
+		docs/bugs.md
+		docs/formats.md
+		docs/sndfile_info.md
+		docs/new_file_type_howto.md
+		docs/win32.md
+		docs/FAQ.md
+		docs/lists.md
+		docs/embedded_files.md
+		docs/octave.md
+		docs/tutorial.md
+		)
+	install (FILES ${dist_doc_DATA} DESTINATION ${CMAKE_INSTALL_DOCDIR})
+
 endif ()
-set (dist_doc_DATA
-	docs/index.md
-	docs/libsndfile.jpg
-	docs/libsndfile.css
-	docs/print.css
-	docs/api.md
-	docs/command.md
-	docs/bugs.md
-	docs/formats.md
-	docs/sndfile_info.md
-	docs/new_file_type_howto.md
-	docs/win32.md
-	docs/FAQ.md
-	docs/lists.md
-	docs/embedded_files.md
-	docs/octave.md
-	docs/tutorial.md
-	)
-install (FILES ${dist_doc_DATA} DESTINATION ${CMAKE_INSTALL_DOCDIR})
 
 if (INSTALL_PKGCONFIG_MODULE)
 	install (FILES ${CMAKE_CURRENT_BINARY_DIR}/sndfile.pc DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig)
